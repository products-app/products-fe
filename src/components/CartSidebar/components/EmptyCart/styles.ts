export default {
  container: 'flex justify-center flex-col items-center h-5/6 overflow-hidden',
  defaultImg: 'w-full px-8',
  message: 'text-gray500',
}
