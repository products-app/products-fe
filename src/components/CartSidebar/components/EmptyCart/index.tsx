import { Text } from '@lebernardo/react'
import { defaultImages } from '@/constants/images'
import styles from './styles'

const EmptyCart = () => (
  <div className={styles.container}>
    <img
      className={styles.defaultImg}
      src={defaultImages.emptyCart}
      alt="Imagem de carrinho vazio"
    />
    <Text className={styles.message}>Adicione algo em seu carrinho...</Text>
  </div>
)

export default EmptyCart
