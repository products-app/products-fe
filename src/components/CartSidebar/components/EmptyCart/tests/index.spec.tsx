import { describe, it, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import EmptyCart from '../index'

describe('Empty Card component', () => {
  it('should render correctly', () => {
    const { container } = render(<EmptyCart />)
    expect(container).toMatchSnapshot()
  })

  it('should contain the empty cart message', () => {
    const { getByText } = render(<EmptyCart />)
    expect(getByText('Adicione algo em seu carrinho...')).toBeDefined()
    expect(() => getByText('no exist')).toThrow()
  })

  it('should contain the empty cart image', () => {
    render(<EmptyCart />)
    const emptyCardImage = screen.getByRole('img')
    expect(emptyCardImage).toHaveAttribute('src', '/empty_cart.png')
    expect(emptyCardImage).toHaveAttribute('alt')
  })
})
