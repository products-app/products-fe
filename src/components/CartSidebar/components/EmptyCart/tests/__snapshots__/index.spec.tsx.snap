// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Empty Card component > should render correctly 1`] = `
<div>
  <div
    class="flex justify-center flex-col items-center h-5/6 overflow-hidden"
  >
    <img
      alt="Imagem de carrinho vazio"
      class="w-full px-8"
      src="/empty-cart.png"
    />
    <p
      class="c-dgMvMW c-dgMvMW-kWyXas-size-md text-gray500"
    >
      Adicione algo em seu carrinho...
    </p>
  </div>
</div>
`;
